### ISSUE TITLE
gc on Windows

### ISSUE THREAD

**Issue by achingbrain** [ipfs/go-ipfs#7225]
#### Version information:

```
go-ipfs version: 0.4.23-
Repo version: 7
System version: amd64/windows
Golang version: go1.13.7
```

#### Description:

The gc-related tests on Windows seem to fail quite often in CI running against go-ipfs over http, eg:

```console
  interface-ipfs-core tests
    .repo.gc
2020-04-27T17:48:36.259Z ipfsd-ctl:daemon:stdout initializing IPFS node at C:\Users\alex\AppData\Local\Temp\go_ipfs_gHoTPFLNlfxgb48B3mk6w
generating 1024-bit RSA keypair...done
peer identity: QmXK6DYbTQwGZioxPzStnsDWvWNvyZ2PZkXvMSG4uVwpCM
to get started, enter:

        ipfs cat /ipfs/QmS4ustL54uo8FzR9455qaxZwuMiUhyvMcX9Ba8nUH4uVv/readme

2020-04-27T17:48:36.259Z ipfsd-ctl:daemon:stderr
2020-04-27T17:48:36.490Z ipfsd-ctl:utils Unable to open api file
2020-04-27T17:48:36.588Z ipfsd-ctl:daemon:stdout Initializing daemon...
2020-04-27T17:48:36.589Z ipfsd-ctl:daemon:stdout go-ipfs version: 0.4.23-
Repo version: 7
System version: amd64/windows
Golang version: go1.13.7
2020-04-27T17:48:36.643Z ipfsd-ctl:daemon:stdout Swarm listening on /ip4/127.0.0.1/tcp/52987
2020-04-27T17:48:36.643Z ipfsd-ctl:daemon:stdout Swarm listening on /p2p-circuit
2020-04-27T17:48:36.647Z ipfsd-ctl:daemon:stdout Swarm announcing /ip4/127.0.0.1/tcp/52987
2020-04-27T17:48:36.647Z ipfsd-ctl:daemon:stdout API server listening on /ip4/127.0.0.1/tcp/52988
WebUI: http://127.0.0.1:52988/webui
2020-04-27T17:48:36.700Z ipfsd-ctl:daemon:stdout Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/52989
Daemon is ready

(node:2384) ExperimentalWarning: Readable[Symbol.asyncIterator] is an experimental feature. This feature could change at any time
      √ should run garbage collection (85ms)
      √ should clean up unpinned data (132ms)
      √ should clean up removed MFS files (209ms)
      √ should clean up block only after unpinned and removed from MFS (191ms)
      1) should clean up indirectly pinned data after recursive pin removal
2020-04-27T17:48:37.544Z ipfsd-ctl:daemon:stderr 18:48:37.542 ERROR blockstore: blockstore.AllKeysChan got err: walk failed: open C:\Users\alex\AppData\Local\Temp\go_ipfs_gHoTPFLNlfxgb48B3mk6w\blocks\du-213173027: The process cannot access the file because it is being used by another process. blockstore.go:222

  4 passing (2s)
  1 failing

  1) interface-ipfs-core tests
       .repo.gc
         should clean up indirectly pinned data after recursive pin removal:
     AssertionError: expected [ Array(4) ] to include 'QmYUY5XnbdTH914VW6dfGSvuyMuBXdjaHk2aUKNWQxMMNQ'
      at Context.it (C:\Users\alex\Documents\Workspaces\ipfs\js-ipfs\packages\interface-ipfs-core\src\repo\gc.js:215:43)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

Sometimes, but not always, they are accompanied by this sort of message printed to stderr:

```
2020-04-27T17:48:37.544Z ipfsd-ctl:daemon:stderr 18:48:37.542 ERROR blockstore: blockstore.AllKeysChan got err: walk failed: open C:\Users\alex\AppData\Local\Temp\go_ipfs_gHoTPFLNlfxgb48B3mk6w\blocks\du-213173027: The process cannot access the file because it is being used by another process. blockstore.go:222
```

Has anyone else seen this on Windows?

**Comment by Stebalien** [comment]
This should be fixed in go-ipfs 0.5.0. We now create all temporary files in a temporary file directory that shouldn't be walked by the iterator.

**Comment by achingbrain** [comment]
Turns out that error was a red herring. The test failure was a bug in the http client.  Fixed now!
